.circleci/config.yml

```diff
@@ -0,0 +1,74 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: mozilla/cidockerbases:docker-latest
+    working_directory: /
+
+    steps:
+        - run:
+            name: Host info
+            command: uname -v
+
+        - run:
+            name: Install essential packages
+            command: |
+                apt-get update
+                apt-get install pigz
+
+        - checkout:
+            path: /ichnaea
+
+        - run:
+            name: Create version.json
+            working_directory: /ichnaea
+            command: |
+                # create a version.json per https://github.com/mozilla-services/Dockerflow/blob/master/docs/version_object.md
+                printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n' \
+                "$CIRCLE_SHA1" \
+                "$CIRCLE_TAG" \
+                "$CIRCLE_PROJECT_USERNAME" \
+                "$CIRCLE_PROJECT_REPONAME" \
+                "$CIRCLE_BUILD_URL" > /ichnaea/version.json
+
+        - store_artifacts:
+            path: /ichnaea/version.json
+
+        - setup_remote_docker
+
+        - run:
+            name: Get Info
+            command: |
+               docker info
+               which docker-compose
+               docker-compose --version
+
+        - run:
+            name: Build Docker images
+            working_directory: /ichnaea
+            command: |
+                # consolidated from a bunch of commands in the older format
+                I="image-$(date +%j).gz"
+                if [[ -e ~/docker/$I ]]
+                    then echo "Loading $I"
+                    pigz -d -c ~/docker/$I | docker load
+                fi
+                docker build -t ${DOCKERHUB_REPO:-mozilla/location} .
+                mkdir -p ~/docker
+                rm -f ~/docker/*
+                docker save ${DOCKERHUB_REPO:-mozilla/location} | pigz --fast -c > ~/docker/$I
+                ls -l ~/docker
```

Comment:
This is a bunch of interesting code to cache intermediary Docker layers to speed up Circle CI runs. One thing that concerns me is that it presumes that we have `Dockerfile` set up in such a way that it correctly breaks the cache when appropriate. I find that to be a common source of errors and it's tricky to test and verify it's correct.

I would simplify this and nix the layer-caching code. We can re-add it if build times are significant.

Reply:
+1 this caching code is ancient. Simplify first, speedup later.